File: internal/modules/notification/model/announcement.go
// Package model 提供通知公告相关的领域模型
package model

import "time"

// 公告状态常量
const (
	AnnouncementStatusDraft     = 0 // 草稿
	AnnouncementStatusPublished = 1 // 已发布
	AnnouncementStatusOffline   = 2 // 已下架
)

// 公告范围常量
const (
	AnnouncementScopeAll    = "all"    // 全平台
	AnnouncementScopeTenant = "tenant" // 指定租户
)

// Announcement 平台通知公告
type Announcement struct {
	ID             string     `json:"id"`
	Title          string     `json:"title"`
	Content        string     `json:"content"`
	Scope          string     `json:"scope"`            // all / tenant
	TargetTenantID string     `json:"target_tenant_id"` // scope=tenant 时指定租户
	Status         int        `json:"status"`           // 0 草稿 / 1 已发布 / 2 已下架
	PublisherID    string     `json:"publisher_id"`     // 发布人（平台管理员）
	PublishAt      *time.Time `json:"publish_at"`
	ExpireAt       *time.Time `json:"expire_at"`
	CreatedAt      time.Time  `json:"created_at"`
	UpdatedAt      time.Time  `json:"updated_at"`
}

File: web-admin/src/api/modules/announcement.ts
import request from '../request'

export interface AnnouncementItem {
  id: string
  title: string
  content: string
  scope: string
  target_tenant_id: string
  status: number
  status_text?: string
  publisher_id: string
  publish_at: string | null
  expire_at: string | null
  created_at: string
  updated_at: string
}

export interface PaginationMeta {
  page: number
  page_size: number
  total: number
  total_pages: number
}

export interface PaginatedResult<T> {
  data: T[]
  pagination: PaginationMeta
}

export function getAnnouncementList(params?: {
  page?: number
  page_size?: number
  keyword?: string
  status?: number
  scope?: string
}) {
  return request.get<any, PaginatedResult<AnnouncementItem>>('/admin/announcements', { params })
}

export function getAnnouncementDetail(id: string) {
  return request.get<any, AnnouncementItem>(`/admin/announcements/${id}`)
}

export function createAnnouncement(data: {
  title: string
  content: string
  scope: string
  target_tenant_id?: string
  status: number
  publish_at?: string
  expire_at?: string
}) {
  return request.post<any, AnnouncementItem>('/admin/announcements', data)
}

export function updateAnnouncement(id: string, data: {
  title: string
  content: string
  scope: string
  target_tenant_id?: string
  status: number
  publish_at?: string
  expire_at?: string
}) {
  return request.put<any, AnnouncementItem>(`/admin/announcements/${id}`, data)
}

export function updateAnnouncementStatus(id: string, status: number) {
  return request.put<any, AnnouncementItem>(`/admin/announcements/${id}/status`, { status })
}

export function deleteAnnouncement(id: string) {
  return request.delete<any, { id: string }>(`/admin/announcements/${id}`)
}

export function listTenantAnnouncements(params?: { page?: number; page_size?: number }) {
  return request.get<any, PaginatedResult<AnnouncementItem>>('/announcements', { params })
}